src/coreclr/vm/methodtablebuilder.cpp

```diff
@@ -8054,7 +7905,136 @@ VOID MethodTableBuilder::PlaceThreadStaticFields()
 //
 // Place instance fields
 //
-VOID    MethodTableBuilder::PlaceInstanceFields(MethodTable ** pByValueClassCache)
+VOID MethodTableBuilder::PlaceInstanceFields(MethodTable** pByValueClassCache)
+{
+#ifdef _DEBUG
+    LPCUTF8 szName;
+    LPCUTF8 szNamespace;
+    if (FAILED(GetMDImport()->GetNameOfTypeDef(GetCl(), &szName, &szNamespace)))
+    {
+        szName = szNamespace = "Invalid TypeDef record";
+    }
+
+    if (g_pConfig->ShouldBreakOnStructMarshalSetup(szName))
+        CONSISTENCY_CHECK_MSGF(false, ("BreakOnStructMarshalSetup: '%s' ", szName));
```

Comment:
This break seems off. Is this only for marshal field placement?

Reply:
If so, the name of the method should be updated.

Reply:
Back before I started refactoring layout the first time, native and managed layout were done at the same time, so they both hit this. I've maintained that experience since. I can remove it here and keep it only for unmanaged layout.